frontend/components/CellOutput.js

```diff
@@ -683,3 +695,44 @@ export let highlight = (code_element, language) => {
         }
     }
 }
+
+/**
+ * Generates a copy button for Markdown code blocks and copies them into the clipboard.
+ * @param {HTMLElement | null} pre - The <pre> element containing the code block.
+ */
+export const generateCopyCodeButton = (pre) => {
+    if (!pre) {
+        console.error('Error: pre is null.');
+        return;
+    }
+
+    const copyToClipboard = () => {
+        const range = document.createRange()
+        range.selectNode(pre)
+        const txt = range.startContainer.textContent || ''
+        navigator.clipboard
+            .writeText(txt)
+            .then(() => {
+                console.log("Code copied to clipboard:\n" + txt)
+            })
+            .catch((error) => {
+                console.error("Error copying code to clipboard:", error)
+            })
+    }
+
+    // create copy button
+    const copyCodeButton = document.createElement("button")
+    copyCodeButton.className = "markdown-code-block-copy-code-button"
+    copyCodeButton.addEventListener("click", copyToClipboard)
+
+    // Create copy button image
+    const copyImg = document.createElement("img")
+    copyImg.src = "https://unpkg.com/ionicons@7.1.0/dist/svg/copy-outline.svg"
```

Comment:
```suggestion
    copyImg.src = new URL("https://unpkg.com/ionicons@7.1.0/dist/svg/copy-outline.svg", import.meta.url)
```

I think this will help the bundler prepare the offline export, so may be a good addition